internal/impl/devbox.go

```diff
@@ -98,6 +100,16 @@ func Open(path string, writer io.Writer) (*Devbox, error) {
 		plugin.WithLockfile(lock),
 	)
 	box.lockfile = lock
+
+	if box.HasDeprecatedPackages() && !legacyPackagesWarningHasBeenShown {
+		legacyPackagesWarningHasBeenShown = true
+		ux.Fwarning(
+			os.Stderr, // Always stderr. box.writer should probably always be err.
+			"Your devbox.json contains packages in legacy format. "+
+				"Please run `devbox update` to update your devbox.json.\n",
+		)
```

Comment:
I'll test, but not sure what we can do about that. Maybe another reason why envrc should only call devbox once?

Reply:
Even with one shellenv call, we do invoke devbox again in the generated shellenv (i think)

but maybe we can look into that and remove it. IIRC it may be older stuff like `devbox log` which we no longer use or need.

Reply:
Actually, maybe we do need it since we care about perf for shell-start